quickwit-doc-mapping/src/default_doc_mapper/default_mapper.rs

```diff
@@ -0,0 +1,378 @@
+            field_mappings: value
+                .field_mappings
+                .field_mappings()
+                .unwrap_or_else(Vec::new),
+        }
+    }
+}
+
+/// Default [`DocMapper`] implementation
+/// which defines a set of rules to map json fields
+/// to tantivy index fields.
+///
+/// The mains rules are defined by the field mappings.
+#[derive(Serialize, Deserialize, Clone)]
+#[serde(try_from = "DefaultDocMapperBuilder", into = "DefaultDocMapperBuilder")]
+pub struct DefaultDocMapper {
+    /// Store the json source in a text field _source.
+    store_source: bool,
+    /// Default map of (field name, text field) used for search.
+    default_search_fields: HashMap<String, Field>,
+    /// List of field mappings which defines how a json field is mapped to index fields.
+    field_mappings: FieldMappingEntry,
+    /// Schema generated by the store source and field mappings parameters.
+    #[serde(skip_serializing)]
+    schema: Schema,
+}
+
+impl std::fmt::Debug for DefaultDocMapper {
+    fn fmt(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+        formatter
+            .debug_struct("DefaultDocMapper")
+            .field("store_source", &self.store_source)
+            .field("default_search_fields", &self.default_search_fields.keys())
+            // TODO: complete it.
+            .finish()
+    }
+}
+
+#[typetag::serde(name = "default")]
+impl DocMapper for DefaultDocMapper {
+    fn doc_from_json(&self, doc_json: &str) -> Result<Document, DocParsingError> {
+        let mut document = Document::default();
+        if self.store_source {
+            let source = self.schema.get_field(SOURCE_FIELD_NAME).ok_or_else(|| {
+                DocParsingError::NoSuchFieldInSchema(SOURCE_FIELD_NAME.to_string())
+            })?;
+            document.add_text(source, doc_json);
+        }
+        let json_obj: JsonValue = serde_json::from_str(doc_json).map_err(|_| {
+            let doc_json_sample: String = if doc_json.len() < 20 {
+                String::from(doc_json)
+            } else {
+                format!("{:?}...", &doc_json[0..20])
+            };
+            DocParsingError::NotJson(doc_json_sample)
+        })?;
+        let parsing_result = self.field_mappings.parse(&json_obj)?;
+        for (field_path, field_value) in parsing_result {
+            let field = self
+                .schema
+                .get_field(&field_path.tantivy_field_name())
+                .ok_or_else(|| {
+                    DocParsingError::NoSuchFieldInSchema(field_path.tantivy_field_name())
+                })?;
+            document.add(FieldValue::new(field, field_value))
+        }
+        Ok(document)
+    }
+
+    fn query(&self, _request: &SearchRequest) -> anyhow::Result<Box<dyn Query>> {
+        todo!()
+    }
+
+    fn schema(&self) -> Schema {
+        self.schema.clone()
+    }
+}
+
+pub fn _load_config_file(path: &Path) -> anyhow::Result<DefaultDocMapper> {
+    match File::open(path) {
```

Comment:
nitpick: Why the different handling of the File open error?
The io::Result should convert to anyhow::Result without any trouble.
```
let config_file = File::open(path)?;
let reader = BufReader::new(config_file);
let builder: DefaultDocMapperBuilder = serde_json::from_reader(reader)?;
let doc_mapper = builder.build()?;
Ok(doc_mapper)
```

Reply:
agreed